api/v1/chiadatalayer_types.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2024 Chia Network Inc.
+*/
+
+package v1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ChiaDataLayerSpec defines the desired state of ChiaDataLayer
+type ChiaDataLayerSpec struct {
+	CommonSpec `json:",inline"`
+
+	// ChiaConfig defines the configuration options available to Chia component containers
+	ChiaConfig ChiaDataLayerSpecChia `json:"chia"`
+
+	// DataLayerHTTPConfig defines the desired state of an optional data_layer_http sidecar
+	// +optional
+	DataLayerHTTPConfig ChiaDataLayerHTTPSpecChia `json:"dataLayerHTTP"`
+
+	// Strategy describes how to replace existing pods with new ones.
+	// +optional
+	Strategy *appsv1.DeploymentStrategy `json:"strategy,omitempty"`
+}
+
+// ChiaDataLayerSpecChia defines the desired state of Chia component configuration
+type ChiaDataLayerSpecChia struct {
```

Comment:
What settings would be valuable for this? Off the top of my head, it would be cool to be able to add or delete store IDs from the builtin wallet in the yaml config. But doing that I believe would likely require config sidecar be developed that runs alongside the data_layer container in the Pod and makes all the RPC requests for managing stores from the yaml config for a ChiaDatalayer resource. 

Maybe there's another option? But I don't think using the chia-docker entrypoint script works as I believe the data_layer/wallet RPC needs to be up.

Reply:
If we go that far with developing the config sidecar, it may as well support all of the actions from https://docs.chia.net/guides/datalayer-user-guide/#create-a-data-store

Reply:
Speaking with a resident data_layer expert, this is unnecessary. CADT just talks directly to data_layer/wallet RPCs and adds/deletes store IDs as required.